lms/static/js/learner_dashboard/views/program_details_view.js

```diff
@@ -31,6 +35,16 @@
 
                 postRender: function() {
                     // Add subviews
+                    new CollectionListView({
+                        el: '.js-course-list',
+                        childView: CourseCardView,
+                        collection: new Backbone.Collection(this.context.course_codes),
```

Comment:
Sorry, I think there was a mis-communication. It's not the Collection View that I was wanting to be instantiated before render, it's the actual Collection itself.